### ISSUE TITLE
Display short commit SHA of commit links in comment previews located in News Feed

### ISSUE THREAD

**Issue by Tunous** [slapperwan/gh4a#453]
![bad](http://i.imgur.com/gTgC965.png) 

vs

![good](http://i.imgur.com/tlmgjE0.png)

**Comment by maniac103** [comment]
Probably needs some HTML rendering or using the HTML variant of some field.

**Comment by maniac103** [comment]
Looks like at least for this particular event (IssueCommentPayload) there's no HTML variant, so one would need to run those body fields through a markdown parser and then converting the HTML to spans. I've played a bit with textmark in the local-markdown branch, but it has a number of compatibility issues with Github's markdown flavour (e.g. issue #377 resurfaces with it). An alternative would be querying the server to render the markdown (see `MarkdownLoader`), but I don't think it's a good alternative here due to latency and used data volume. Thoughts?

**Comment by Tunous** [comment]
Definitely, additional queries to the server shouldn't be used as it's just waste of data. Another alternative is to write a simple parser just for the GitHub specific links which would handle this.

**Comment by maniac103** [comment]
Yeah, thinking more about it, it's not a matter of markdown parsing. The formatting simply is missing completely. A local replacement by regex would probably work, it's just a matter of catching all URL variants.
